### DISCUSSION TITLE
Unstable PLUMED simulation when using the OpenMM/DeepMD plugin

### DISCUSSION THREAD

**Discussion by varun-go** [deepmodeling/deepmd-kit#3577] [General]
I am attempting an umbrella sampling simulation using Plumed using the OpenMM/DeepMD [plugin](https://github.com/JingHuangLab/openmm_deepmd_plugin). The simulation consists of formic acid in water and I am attempting to simulate a dissociation reaction. The CV for biasing is a form of coordination number and is mentioned in this reference [study](https://pubs.acs.org/doi/full/10.1021/jacs.2c03099) (SI - equation S1).

In an unbiased simulation of the system, there are no issues. However, when I apply the plumed biasing force, the system is stable only for ~1 ps before it explodes. I have run the simulation with a high writeout frequency and observe that the system temperature fluctuates around the set point but then suddenly spikes and increases. I have also tested decreasing the coupling to the temperature bath, but the issue persists. I confirmed that this issue is prevalent even when the force constant in the biasing force is reduced from the original value of 15 eV down to even 1 eV.

I would like to rule out that the issue is due to communication between Plumed/OpenMM/DeepMD. I have checked the other issues in the discussions and don't see a related post. If someone could assist, that would be great.

I have attached the simulation script and model in the following Google drive [link](https://drive.google.com/drive/folders/1oti0ORN5-m8ozweDM-ZfroowtrWc5JW1?usp=sharing).  The script to run is python file and other necessary files are provided as well. One can remove the plumed biasing force by commenting out line 64 which starts with the term `restraint`. 

Please let me know if additional information is needed.

**Comment by wanghan-iapcm** [comment]
have you sampled the dissociation events in your training data? or could you please run the enhanced sampling MD simulations with multiple DP models to calculate the model deviation, which is able detect the when the model fails .

**Reply by varun-go** [reply]
Thank you for your response! Yes, the training data does sample the dissociation event. The trained model was used to study the same dissociation reaction using the LAMPPS/Plumed interface and was reported to be stable there.

**Reply by wanghan-iapcm** [reply]
I would firstly suspect the unstable enhanced sampling MD trajectories run out of the coverage of the training data. A good indication of such events is the model deviation.

**Reply by varun-go** [reply]
Thank you for the response. The issue appeared to be from the openmm-plumed plugin version that I was using (v 2.0)  openmm/openmm-plumed#78. When I upgraded the plugin version to 2.0.1, the system remained stable with the addition of the biasing force.

**Reply by wanghan-iapcm** [reply]
Thank you for the update!
